File: src/org/onion_lang/onion/compiler/FileInputSource.java
/* ************************************************************** *
 *                                                                *
 * Copyright (c) 2005, Kota Mizushima, All rights reserved.       *
 *                                                                *
 *                                                                *
 * This software is distributed under the modified BSD License.   *
 * ************************************************************** */
package org.onion_lang.onion.compiler;

import java.io.IOException;
import java.io.Reader;

import org.onion_lang.onion.compiler.util.Inputs;

public class FileInputSource implements InputSource {
  private Reader reader;
  private String file;
  
  public FileInputSource(String file) {
    this.file = file;
  }
  
  public Reader openReader() throws IOException {
    if(reader == null) reader = Inputs.newReader(file);
    return reader;
  }
  
  public String getName() {
    return file;
  }
}

File: src/main/scala/onion/compiler/FileInputSource.scala
/* ************************************************************** *
 *                                                                *
 * Copyright (c) 2005-2012, Kota Mizushima, All rights reserved.  *
 *                                                                *
 *                                                                *
 * This software is distributed under the modified BSD License.   *
 * ************************************************************** */
package onion.compiler

import java.io.IOException
import java.io.Reader
import onion.compiler.toolbox.Inputs

class FileInputSource(val name: String) extends InputSource {
  private lazy val reader: Reader = Inputs.newReader(name)

  def openReader: Reader = reader
}